### DISCUSSION TITLE
Medusa 2.0 Public Preview Release

### DISCUSSION THREAD

**Discussion by olivermrbl** [medusajs/medusa#7955] [Announcements]
Today, we’re excited to share a Preview Release of Medusa 2.0. 

The following document covers resources, guides, and things to know about the first preview release of Medusa 2.0. It will help you get started and give you a high-level overview of your new Medusa 2.0 application and what’s still to be done.

> **The preview release is not ready for production and is only intended for experimentation and development of new projects.**

The release candidate later this year will include more fully-fledged resources for 2.0, including an overview of breaking changes, migration guides, and much more that will ensure a smoother onboarding.

**We advise against upgrading your project to 2.0 at this point,** as we are still working on documentating the best possible path.

These release notes are only intended to help you start and introduce you to your new Medusa 2.0 project. For a more in-depth exploration of the major upgrade, refer to [our new documentation](https://docs.medusajs.com/v2).

We are publishing this release to solicit early user and community feedback. Please file Issues for bug reports and/or submit Discussions with input or questions. 

We thank you all very much in advance.

---

## Get started

To get started using the preview release, run the following command:

```bash
npx create-medusa-app@preview
```

This command will create a new Medusa project with our redesigned admin and a 2.0-compatible Next.js storefront. The Medusa application and the Next.js storefront are separate projects in separate folders.

The folder structure of Medusa projects has changed:

```diff
my-project
 ├── src
 │   ├── admin
 │   ├── api
 │   ├── jobs
 │   ├── subscribers
-│   ├── services
-│   ├── models
-│   ├── migrations
-│   ├── loaders
+│   ├── modules
+│   │   └── my-module
+│   │       ├── loaders
+│   │       ├── migrations
+│   │       ├── models
+│   │       ├── index.ts
+│   │       └── service.ts
+│   ├── workflows
 └── medusa-config.ts
```

The `modules` folder replaces a lot of folders from a 1.0 project and will contain most business logic customizations in your Medusa 2.0 application. We highly encourage you to read the [section of our docs covering this new concept in detail](https://docs.medusajs.com/v2/basics/modules-and-services).

The `workflows` folder holds your custom workflows. Workflows are a fundamental concept in cross-module operations and a core element of Medusa 2.0. Read more about workflows [here](https://docs.medusajs.com/v2/basics/workflows).

Additionally, `medusa-config.js` has been cleaned up and improved with a new type-safe utility function for defining the config for your Medusa project. The file can also now live as both a JS and TS file.

```ts
export default defineConfig({
  projectConfig: {
    http: {
      storeCors: "<STORE_CORS>",
      adminCors: "<ADMIN_CORS>",
      authCors: "<AUTH_CORS>"
    },
    redisUrl: "<REDIS URL>",
    databaseUrl: "<DATABASE_URL>",
  },
  admin: {
    backendUrl: "https://my-medusa-server.medusajs.app",
  },
})

```

You are now ready to explore all the new features in the preview release. However, we recommend reading through our [documentation](https://docs.medusajs.com/v2/) to fully understand what is new in 2.0 and how it differs from 1.0.

Additionally, you can find references and code snippets for our tools in our [Learning Resources](https://docs.medusajs.com/v2/resources).

---

## Keeping up with preview releases

We release preview versions every three hours. We recommend setting the version of your Medusa dependencies to `preview`, so you can regularly update without knowing specific version numbers.

All your **Medusa** dependencies in `package.json` should look something like this:

```json
{
  "dependencies": {
    "@medusajs/medusa": "preview",
    "@medusajs/pricing": "preview",
    "@medusajs/product": "preview",
    ...
  }
}
```

To ensure an upgrade to a new version is completed correctly, run the following sequence of commands:

```tsx
rm -rf node_modules
rm yarn.lock // or package-lock.json

yarn // If you are using yarn berry, you need to create the lock-file first
```

---

## What’s new?

Medusa 2.0 is a huge upgrade, in fact, the biggest upgrade since we open-sourced the project four years ago. The upgrade comes with many new features and improvements, some of which are still work-in-progress. In the following sections, we will briefly cover these features and the parts of the product still subject to change. A more detailed changelog will be released as part of the release candidate and, subsequently, the official release later this year.

### Admin dashboard

Our admin dashboard has been completely overhauled, improving its overall look and feel to provide end-users with a more cohesive and consistent experience. 

Additionally, we have migrated away from Webpack to Vite, which improves the developer experience with features like HMR and allows us and our users to tap into a rich and flourishing community and plugin ecosystem.

Read more about the admin dashboard and its extension capabilities [here](https://docs.medusajs.com/v2/advanced-development/admin).

A few admin flows have yet to be built. This is primarily around returns, exchanges, claims, and order edits. We expect to finalize these in the coming weeks.

 

### Modules architecture

Our modules foundation has been re-architected to entirely decouple modules from one another, enabling greater composability, a new standalone usage mode, and a more approachable incremental adoption of Medusa. 

Read more about the architecture refactor in our [2.0 announcement from earlier this year](https://medusajs.com/recap/).

**New commerce features**

We saw reworking the module's architecture as an opportunity to rethink the feature set of some of the business domains in Medusa 1.x. As a result, we are excited to introduce a range of advanced capabilities across a series of modules. Here are a few noteworthy improvements:

- Stock location: Build omnichannel experiences with new stock location features, including “Buy online, pick up in-store” options.
- Inventory: Compose multi-part variants and share inventory across products with our new inventory management features
- Authentication: Enable authentication with external providers, such as Google, with our new authentication and user management features
- Promotion: Create granular promotions with our new rules engine. Use conditions to create bundled discounts, buy-get, spend-get, temporary campaigns, and more

Many other modules have also seen improvements. Browse through all of them in our [learning resources](https://docs.medusajs/v2/resources).

### Data modeling language

A core feature of Medusa is the ability to extend the default commerce application with additional business logic to cater to a specific use case, natively integrate with third-party providers, replace existing domains, and more. 

At the core of extensions in Medusa 2.0 are modules, and at the core of modules are data models. Data models define the concepts living within your application. In our commerce modules, you find orders, products, customers, promotions, etc. By introducing your own data models, you extend this semantic layer of your application, enabling you to build features and custom workflows by integrating existing and new concepts. Because data models are at the core of every customization, we decided to make them easier to work with. 

We are excited to announce a new data modeling language in Medusa 2.0 that will improve the experience of defining and working with data in your application. You can read more about the new feature [here](https://medusa-docs-v2-git-docs-v2-medusajs.vercel.app/basics/data-models).

Below is a simple example of what data models will look like in 2.0:

<div align="center">
<img src="https://github.com/medusajs/medusa/assets/59018053/35640fdc-0988-43b6-9c08-3de2af55c3d6" width="500" />
</div>

### Long-running Workflows

We are excited to extend our existing workflow capabilities with Long-running Workflows for processing long-running operations in the background. Instead of synchronously receiving a workflow's result, Long-running Workflows are subscribed to for status changes and execution progress. Additionally, the Long-running Workflows comes with an API for programmatically moving from one step to the next. These features make Long-running Workflows a great fit for use cases that require a human-in-the-loop or operations with a varying long-running execution time. 

For example, we use Long-running Workflows extensively to build the infrastructure provisioning of Medusa Cloud. 

Alongside the Long-running Workflows, we are also introducing a new admin section for managing them. Here, you can see currently executing and completed workflows, providing full transparency into your long-running operations.

<div align="center">
<img src="https://github.com/medusajs/medusa/assets/59018053/4d6854d3-508c-4dad-9b28-e453ee636fc5" width="700" />
</div>

Medusa 2.0 comes with much more than what is highlighted in this post. We will cover everything in-depth in the official release notes.

---

These release notes are published as a GitHub Discussion as well for comments and questions on the preview release.

**Comment by HarleySalas** [comment]
### Some minor bugs and criticisms

### Bugs

- [Admin Dashboard] Pressing `cmd + K` opens google search as well as dashboard search on chrome
- [Admin Dashboard] On a product page, adding a filter for variants, suchas " managed inventory = true or false, you get an error page. (with default seed)
- [Admin Dashboard] When inviting a user, their email isn't prefilled in the signup form and they can enter any email. I believe it should have their email automatically prefilled and the input disabled.
- [Storefront] When Meilisearch is not enabled (by default), it shows the search feature anyways and results in an error when attempting to search. Very minor thing, but it was seemingly lost between v1 and v2.

### Criticisms

Darkmode Save/Complete Buttons look *disabled*. - I recommend making them a bit brighter, so it's clear that they are pressable.


`/app/products/{productId}/variants/{variantId}/manage-items` should allow you to create an inventory item while in the modal, rather than forcing you to leave the product creation route.


In the same route as the previous criticism (manage-items), when you click the `Item` field to search, it should give some indication that they should start typing, or prefill a few options in the dropdown. If you simply click it, it feels like there is nothing to search and that its just *empty*.


**PLEASE SAVE ME FROM THE WHITE SCROLLBARS** (chrome, linux mint)
![image](https://github.com/medusajs/medusa/assets/38236585/e30f2d7c-38c2-49a2-9302-a67048f11330)


If no attribute is selected, the "Select Operator" and "Select Values" fields should be disabled.
![image](https://github.com/medusajs/medusa/assets/38236585/f4cc0fc2-aeb1-483d-b7e7-f893d5376089)


I want to start the next one off by saying how great I think the implementation of inventory items is. Bundles/crazy variants are **effortless** and is the best way I've seen it done, period.

Now that I've said something nice... I'm going to be a bit harsh now when it comes to the promotions conditions.
![image](https://github.com/medusajs/medusa/assets/38236585/83d84cf1-290e-4df9-9b6b-67fb3910c262)

I think it needs *a lot* of work. If I, a developer that is familiar with SQL cannot comprehend how to use it, how is a regular user supposed to? It also doesn't seem very flexible.

Lets consider a few very common cases where someone would want to have a promotion;

If someone buys `Medusa T-Shirt` and `Medusa Shorts` together, I want to automatically apply a 20% discount to those two items' combined price. How can I do that? What if we want to include more than 1 product in the condition?

What if I don't want to include **any** product in the condition? For example; a promotion, where if you buy 10 of any item, you get a discount.

What about if you want to target a promotion, for the event that a customer spends a specific amount of money? For example; spend more than $1,000 and get 20% off, or something like that?

Theres just  countless scenarios, where it doesn't feel like these conditions are enough? Yet, they seem extremely confusing as well.

### On to the next

Is there still no easy way to set some permissions for users? even at a very basic level? Why should an employee doing fulfillment be able to delete regions, or destroy the store so easily? At a *bare minimum* the only route under settings that should be visible for a normal "user" is `/app/settings/profile`. A special admin/manager role should be required to access the rest there.

 I think these multi-select inputs are very awkward and unfamiliar. I believe it's better to have 2-3 badges, followed by "+3 more..." or something along those lines.
![image](https://github.com/medusajs/medusa/assets/38236585/634189b6-5007-40eb-9c31-598114ddeea1)

Still no variant images by default? It's such a simple feature, yet necessary for so many stores.

Now definitely the most opinionated thing that is certainly *just* preference... I think the contrast between the black and dark gray is too much. I'd rather see the black color come closer to the gray, or the gray closer to the black. (I think moving the black closer to the gray would be best) A tiny criticism, but I'd just say, to me at least, they don't blend in together very well.
![image](https://github.com/medusajs/medusa/assets/38236585/cbe6ae52-bb0c-4c67-8b5c-c867d6ff3486)



I know it was long, but I love Medusa and I believe criticism is important. Thanks to anyone that read through it all. Everything looks like its headed in an amazing direction and I'm excited to start building with 2.0

**Reply by HarleySalas** [reply]
Also, is there any idea of when more than just English will be available in the Dashboard?

**Reply by sradevski** [reply]
Hey @HarleySalas, first of all, I'd like to thank you for the very detailed writeup, this sort of feedback is what can help us build a much better product!

There isn't much to be said on the bug reports, we'll look into each individually and get them fixed. We'll take the design comments you had and try to improve on those 👍 

Regarding users and permissions, what we've done in v2 is paved the way to support role-based access, but we haven't implemented it yet. Now, whether this will come before the full release or some time after, I am not sure yet, but we have it on our radar.

Regarding the media being associated with a variant, I agree with you that this is a very important feature and it's something we'll likely look into adding before the full release.

On promotions, our focus until now was to ensure that dynamic cases such as the ones you said can be supported by our backend, and we intentionally did a very simple FE implementation. This (and pretty much everything else) will be revised and improved in the upcoming few months, and your feedback here is invaluable to guiding that discussion.

Localization is already supported on the admin, we just haven't done the translations in different languages yet. This will be easy to do by the community as well, but we'll likely encourage that once the admin changes at a slower pace than it is today.

Once again, thank you for being vocal and critical, this is the type of feedback we want to hear to do better.

**Reply by HarleySalas** [reply]
Thanks for the response. 

Does this mean that in theory, if I were to provide the localizations I need, they could be merged relatively soon and potentially available shortly after its done?

If so, if you point to the folder that they're done in, I'll get it done ASAP.

**Reply by sradevski** [reply]
@HarleySalas appreciate the eagerness to contribute! Since admin (and the translation file) still goes under a lot of changes (sometimes breaking), we prefer waiting for contributions once we are closer to the full release date. We'll keep you updated when it's the time to add them 👍

**Comment by SalahAdDin** [comment]
**BUG**

At least in Manjaro Linux with Firefox, after successfully initializing the Medusa project, an Admin Panel tab with an invitation/creating admin user screen is not automatically opened on the browser.

**Reply by sradevski** [reply]
@SalahAdDin can you please open this as a Github Issue instead, so we can track it and assign it more easily? Thanks!

**Reply by SalahAdDin** [reply]
@sradevski [done](https://github.com/medusajs/medusa/issues/7987)!

**Reply by renanboni** [reply]
@sradevski it still doesn't up and I can't access the admin dashboard (localhost:7000 or 7001). Do you know what might be the reason?

I get this message from the browser:

**Access to localhost was denied
You don't have authorization to view this page.
HTTP ERROR 403**

However, there are no stack traces or logs. I've been stuck for quite some time this week. I'd love some input here.

Thanks

**Comment by zhaowy3131** [comment]
ecommerce could encourage individual sales to make profit by linking orders related with himself/herself. So, every order could be distinguished who are the sales. Furthermore, ecommerce could also encourage sales to help each other by profit share between sales.
I hope medusa could have these features.

**Reply by HarleySalas** [reply]
Could you try to reword this a bit? It's a bit hard to understand exactly what you're saying.

Are you talking about a feature that would associate a sales representative to a particular order? If so, I'd say that is a bit unusual for e-commerce and sounds more like something you'd want in a point of sale system.

**Comment by JohnGemstone** [comment]
Does this allow for serverless environments?

**Comment by zhaowy3131** [comment]
Yes, just add the sales guy in the order table. It is not only used in
physical stores. If I am the salesperson and I promote the online store
with my specific link on facebook and other social media and clients hit my
link and make an order, then it will be the order related to me.

Furthermore, medusajs could also encourage salesperson to help each other
by sharing profit.

Thanks

On Fri, Jul 5, 2024 at 2:59 AM Harley Salas <
***@***.***> wrote:

> Could you try to reword this a bit? It's a bit hard to understand exactly
> what you're saying. Are you talking about a feature that would associate a
> sales representative to a particular order? If so,
> *DuckDuckGo* removed one tracker. More
> <https://duckduckgo.com/-hGysyZFupsqvAi0nzhKiHW-qw9P256StfIfMCuF4D2dv96dOfKhqznEcl_2WMbbFYaGxLCFPaQOGKtGAFxGrakRjxkEjszLTtzY_bANpBiwFNe8nBsFIzk10dhOeeAmpTCooKQQMPfKAB7DUtTqBBY9RInVDAFFnsjhJIw8Xd-37_et8in-ofUZb4DjHnR6FV-QHEjhO22f7S5Hg8L-4kp1-8jd3T5TCEAPC-l4GP5woZT7J9VrlNsaAHv0cBCMrPgcByv5FpC6LvYxiozR5t6JpIJdoJ8QH1TDIATydeQSVSowxX3mtC4Vjm-ah13AcLoJDLMgklT30cEM46EtqaLNDXbAlogxKrMG0pa1NpqKp8EmJpTuHA-H-BZ7gj-qtKdmPz_VGziE6y-B19KV2YIPuy-1bZnqwEGzXrt5uSEGTuxD7_Hl3I8rIdxRIbuTbZG1T9wNfo9BDUe-_CqaXdErneG5a8Tgz-snuzwPTvX0GS4nJ9JwMyLoxSHS2-KnEVcsiG3EEH>
> Report Spam
> <https://duckduckgo.com/-hGysyZFupsqvAi0nzhKiHW-qw9P256StfIfMCuF4D2dv96dOfKhqznEcl_2WMbbFYaGxLCFPaQOGKtGAFxGrakRjxkEjszLTtzY_bANpBiwFNe8nBsFIzk10dhOeeAmpTCooKQQMPfKAB7DUtTqBBY9RInVDAFFnsjhJIw8Xd-37_et8in-ofUZb4DjHnR6FV-QHEjhO22f7S5Hg8L-4kp1-8jd3T5TCEAPC-l4GP5woZT7J9VrlNsaAHv0cBCMrPgcByv5FpC6LvYxiozR5t6JpIJdoJ8QH1TDIATydeQSVSowxX3mtC4Vjm-ah13AcLoJDLMgklT30cEM46EtqaLNDXbAlogxKrMG0pa1NpqKp8EmJpTuHA-H-BZ7gj-qtKdmPz_VGziE6y-B19KV2YIPuy-1bZnqwEGzXrt5uSEGTuxD7_Hl3I8rIdxRIbuTbZG1T9wNfo9BDUe-_CqaXdErneG5a8Tgz-snuzwPTvX0GS4nJ9JwMyLoxSHS2-KnEVcsiG3EEH>
>
> Could you try to reword this a bit? It's a bit hard to understand exactly
> what you're saying.
>
> Are you talking about a feature that would associate a sales
> representative to a particular order? If so, I'd say that is a bit unusual
> for e-commerce and sounds more like something you'd want in a point of sale
> system.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/medusajs/medusa/discussions/7955#discussioncomment-9966275>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/A2RH4BOULQYKYHMRUBQ5SI3ZKZN7JAVCNFSM6AAAAABKLZXMROVHI2DSMVQWIX3LMV43SRDJONRXK43TNFXW4Q3PNVWWK3TUHM4TSNRWGI3TK>
> .
> You are receiving this because you commented.Message ID:
> ***@***.***>
>


-- 
Thanks
Weiyi Zhao

**Comment by zhaowy3131** [comment]
I have no answer, not familiar with medusa.

On Fri, Jul 5, 2024 at 3:15 AM Jim Johnston <
***@***.***> wrote:

> Does this allow for serverless environments? —Reply to this email
> directly, view it on GitHub, or unsubscribe.You are receiving this because
> you commented.Message ID:
>
> *DuckDuckGo* removed one tracker. More
> <https://duckduckgo.com/-hGysyZFupsqvAi0nzhKiHW-qw9F2umfIVMhtwnW7uOAiu8O_Tnyoas5xHJf9ljG2xWGhsSwhT2kDhirRgBcRq2pEY8ZBI7My07c2P2wDaQYsBTXvJwbBSM5NdHYTnngJqUwqKCkEDD3ygAew1LU6gQWPUSJ1QwBRZ7I4SSMPF3ft-_3rfIp_qH0k8qbwTVyggsumA1WuuU01PANli92Q4nn5RKj27ia4dB4C1NTXKFuEi7ThtvGLpx4Dq7GwJwF22NwJ6t5OoMsDb2pqCXDQr7c4BM96liA-oZ5IBeJrjEVQiNcpw5b0mFI5lno9ax32wAAq5LJNQ8tTHAeGsI6GtyYK8Z0sgG5RYhWlLWZtKQ1XlkxBLcwqM0v8FnuGO6K8q2Y3N90fNIjrJ4nd8xCIrmYo231pWx5acVbBVafqJSJQM2-vfowtZXrajWGIj1yZ7g6of-BqdHoJ6782-nXHLkDMHYc8JE4O9bJ4sMP37V5CkuFzfyYCMC6NUR4uvSlyFHLIhd9AB>
> Report Spam
> <https://duckduckgo.com/-hGysyZFupsqvAi0nzhKiHW-qw9F2umfIVMhtwnW7uOAiu8O_Tnyoas5xHJf9ljG2xWGhsSwhT2kDhirRgBcRq2pEY8ZBI7My07c2P2wDaQYsBTXvJwbBSM5NdHYTnngJqUwqKCkEDD3ygAew1LU6gQWPUSJ1QwBRZ7I4SSMPF3ft-_3rfIp_qH0k8qbwTVyggsumA1WuuU01PANli92Q4nn5RKj27ia4dB4C1NTXKFuEi7ThtvGLpx4Dq7GwJwF22NwJ6t5OoMsDb2pqCXDQr7c4BM96liA-oZ5IBeJrjEVQiNcpw5b0mFI5lno9ax32wAAq5LJNQ8tTHAeGsI6GtyYK8Z0sgG5RYhWlLWZtKQ1XlkxBLcwqM0v8FnuGO6K8q2Y3N90fNIjrJ4nd8xCIrmYo231pWx5acVbBVafqJSJQM2-vfowtZXrajWGIj1yZ7g6of-BqdHoJ6782-nXHLkDMHYc8JE4O9bJ4sMP37V5CkuFzfyYCMC6NUR4uvSlyFHLIhd9AB>
>
> Does this allow for serverless environments?
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/medusajs/medusa/discussions/7955#discussioncomment-9966410>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/A2RH4BOI2V3FT5VHZMSOWATZKZP23AVCNFSM6AAAAABKLZXMROVHI2DSMVQWIX3LMV43SRDJONRXK43TNFXW4Q3PNVWWK3TUHM4TSNRWGQYTA>
> .
> You are receiving this because you commented.Message ID:
> ***@***.***>
>


-- 
Thanks
Weiyi Zhao

**Comment by volfadar** [comment]
Does Medusa 2.0 improve the developer experience (DX) for digital product e-commerce stores? 

Currently, it is still difficult to create digital products in Medusa. Even when I successfully configure Medusa to accept digital products, there are still UI issues. Many unnecessary fields appear in the UI, which are not relevant to the digital product ecosystem. My only solution so far has been to hide these fields with CSS. 

This was the main reason I decided to switch back to WooCommerce, as 90% of my clients sell digital products. While Medusa has many advantages over WooCommerce, such as better customization options and a more beautiful UI, the poor DX for digital products outweighs these benefits. This is very disappointing.

**Comment by xAzoom** [comment]
When will Medusa 2.0 be production ready? Is 3 months realistic?

**Reply by SalahAdDin** [reply]
Minimum three months, maybe six

**Reply by sradevski** [reply]
Production-readiness is a range, not a boolean 😄 I think you will be able to have a production setup in 3 months. There could be some small bugs here and there, and some minor features missing, but I do think in 3 months you should be able to comfortably go to production.

**Comment by arimtiaz** [comment]
What is the stage of building marketplaces using Medusa 2.0?

**Reply by lacymorrow** [reply]
I'd love a write-up as good as https://blog.perseides.org/

**Reply by arimtiaz** [reply]
same @lacymorrow

**Comment by renequez** [comment]
How hard do you believe upgrading from v1 will be? Considering how much the architecture is changing, how will it be possible to migrate custom implementations (workflows, data models etc.)?

**Comment by HarleySalas** [comment]
### Feedback

The `widgets` and `routes` feature for `admin` is cool.
However, I do believe the placement for widgets should be a bit more flexible, particularly for the `products` placement.

For example, perhaps you could add some sort of placement like `products.details.media.before`, so that we can place widgets here:
![image](https://github.com/user-attachments/assets/1c1a0839-b703-4492-836b-e1d3e78e9710)

Currently, it feels like every custom widget is an afterthought and in many cases, the widgets we add may be very important. I think we should be able to weave in to any "zone" there.

**Reply by sradevski** [reply]
Hey @HarleySalas thanks for the feedback!

As we are getting closer to the full release, we will be adding more injection zones, as well as ensuring consistency and flexibility with the existing ones.

cc @kasperkristensen

**Reply by SalahAdDin** [reply]
@sradevski  Are you willing to add videos to the product media?

**Reply by sradevski** [reply]
I'll open a ticket internally and we'll decide if it makes sense.

**Comment by cseajay009** [comment]
Hi @olivermrbl ,

Do we or in future will have capability to deploy medusa backend in microservices based architecture, Suppose we need to scale product api's to speed up product view response time, So in this scenario currently will have to scale the entire backend application, Please throw some light on this topic.


Thanks & Regards
Ajay Singh

**Reply by carlos-r-l-rodrigues** [reply]
Hi @cseajay009,
yes, this is planned to exist in the future. You will be able to deploy independently a module or a set of modules and the main Medusa application will reference them as external modules. So you can scale everything independently.

**Comment by princieneoshore** [comment]
Hi @olivermrbl ,

It would be great too if we can add filters on link, for example in this code, i can only add filters using my_custom fields but it is impossible to add filters using product fields
![image](https://github.com/user-attachments/assets/299d2296-ba25-4b49-8415-3d643b5f1289)

**Reply by carlos-r-l-rodrigues** [reply]
Cross-module filtering/sort will be available in the near future as another module.

**Reply by princieneoshore** [reply]
@carlos-r-l-rodrigues @olivermrbl 
how to use db transaction with medusa js 2.0, i want operations to rollback if one failed

**Reply by olivermrbl** [reply]
Can you elaborate on how and where you want to use the transaction? The usage depends on the context.

**Reply by princieneoshore** [reply]
@olivermrbl  i created a service function and the are multiple create and update operations there that i want to wrap inside transaction

**Comment by princieneoshore** [comment]
@olivermrbl  i also tried to query customer with an email, it shoud be case insensitive

**Comment by kritzware** [comment]
Any updates on when V2 will be production ready?

**Reply by SalahAdDin** [reply]
I saw 23 October.

**Comment by raiyansarker** [comment]
is there any update on migration guide?

**Reply by SalahAdDin** [reply]
Did you check the official documentation?

**Reply by michsko544** [reply]
Is there any migration guide from v1 to v2 available in the documentation yet?

**Reply by raiyansarker** [reply]
> Did you check the official documentation?

Yeah I checked, couldn't seem to find one, maybe I missed but have you seen one?

**Reply by michsko544** [reply]
@raiyansarker @SalahAdDin it seems that something has started happening regarding migration to v2 https://medusajs.com/v2-migration-support/
